openedx/core/djangoapps/external_user_ids/models.py

```diff
@@ -36,3 +43,63 @@ class ExternalId(TimeStampedModel):
     external_id_type = models.ForeignKey(ExternalIdType, db_index=True, on_delete=models.CASCADE)
     user = models.ForeignKey(User, db_index=True, on_delete=models.CASCADE)
     history = HistoricalRecords()
+
+    class Meta:
+        unique_together = (('user', 'external_id_type'),)
+        app_label = 'external_user_ids'
+
+    @classmethod
+    def user_has_external_id(cls, user, type_name):
+        """
+        Checks if a user has an ExternalId of the type_name provided
+        Arguments:
+            user: User to search for
+            type_name (str): Name of the type of ExternalId
+        Returns:
+            (Bool): True if the user already has an external ID, False otherwise.
+        """
+        try:
+            type_obj = ExternalIdType.objects.get(name=type_name)
+        except ExternalIdType.DoesNotExist:
+            LOGGER.info('No external id type of {}'.format(type_name))
+            return False
+        if cls.objects.filter(
+            user=user,
+            external_id_type=type_obj
+        ).exists():
+            return True
```

Comment:
If the goal here to log incorrect type_names, then this can stay as is. If you want to simplify it, Django allows filtering on fields of associated models (foreign keys) so you can do things like 
```
cls.objects.filter(
    user=user,
    external_id_type__name=type_name
).exists()
```
and skip the initial try block. Also 1 DB call instead of 2 on every enrollment. Up to you.

Reply:
See https://docs.djangoproject.com/en/3.0/topics/db/queries/#retrieving-specific-objects-with-filters for reference